web: Expand 'max_execution_duration' type in extension.

iwannabethedev opened the pull request:
The option 'max_execution_duration' previously only supported the type '{secs: number, nanos: number}'. Now it also supports using floating point numbers (and integers).

See the discussion in #10787 and this gist: https://gist.github.com/Dinnerbone/e9b4c6926dd5dc380a4b12c9c762c9bb .

I have only tested it manually a bit, but I did test a few different cases, including that it correctly gave errors on at least one instance of incorrect input, and that it works for both floats and the type `{secs: number, nanos: number}` (and integers as well).

iwannabethedev commented:
@relrelb @Dinnerbone I have not yet added a `console.warn` call when the old deprecated duration format with `{secs: number, nanos: number}` is used; I imagine that this call should be added to the Rust code, but I do not know how to do that.

Do any of you know how to do that in Rust, or should it just be done in the Typescript code? Maybe around this line?: https://github.com/ruffle-rs/ruffle/blob/7af0a17d83ac1506fe37edea4f70bc0b373f06ed/web/packages/core/src/ruffle-player.ts#L715

n0samu commented:
You can do it in the TypeScript code near a similar warning (added in #9802):

https://github.com/ruffle-rs/ruffle/blob/7af0a17d83ac1506fe37edea4f70bc0b373f06ed/web/packages/core/src/ruffle-player.ts#L456-L459

iwannabethedev commented:
@relrelb @Dinnerbone @n0samu I believe this is ready for review. I fixed linting, and ran the auto-formatting for Rust, merged, and fixed the obsolete default values. I have tested the changes manually a little bit.

n0samu commented:
Awesome, thanks! The only problem I notice at a glance is that `Obsolete_Duration` doesn't need an underscore. (Same with isObsolete_Duration, etc)

iwannabethedev commented:
It might make best sense to remove the underscore - I mostly added it in order to make it stand out, since the type is obsolete and its format should not be used. But it does not follow the style guideline, I believe. I will let it be up to you whether I should remove it.

Dinnerbone commented:
I agree with removing the underscore, otherwise this looks good. Thank you!

iwannabethedev commented:
@Dinnerbone I have renamed the type and function.

A test failed for "Test Node.js 20 / Rust stable / ubuntu-22.04", but I did not understand by skimming through the test output what had failed, and it appears that other PRs have also had that test failing.

iwannabethedev commented on web/packages/core/src/ruffle-player.ts:
```diff
@@ -452,6 +453,17 @@ export class RufflePlayer extends HTMLElement {
+        if (
+            this.loadedConfig &&
+            this.loadedConfig.maxExecutionDuration &&
+            isObsoleteDuration(this.loadedConfig.maxExecutionDuration)
```
The type guard function was only used in one place, so I agree with removing it. I have tested that it still works for both the obsolete format and the new format (and gives the warning only for the obsolete format).

Dinnerbone commented:
Can you please rebase this onto master? It has merge conflicts apparently (but github isn't telling me what they are, oddly)

Dinnerbone commented:
That was a merge, can you please try to rebase or would you like me to do it?
